goon.go

```diff
@@ -69,14 +77,72 @@ type Goon struct {
 	KindNameResolver KindNameResolver
 }
 
-// Versioning, so that incompatible changes to the cache system won't cause problems
-var memcacheKeyPrefix = fmt.Sprintf("g%X:", serializationFormatVersion)
-
-// MemcacheKey returns key string of Memcache.
+// MemcacheKey returns the string form of the provided datastore key.
 var MemcacheKey = func(k *datastore.Key) string {
-	return memcacheKeyPrefix + k.Encode()
+	return k.Encode()
 }
 
+// Versioning, so that incompatible changes to the cache system won't cause problems
+var cacheKeyPrefix = fmt.Sprintf("g%X:", serializationFormatVersion)
+
+// cacheKey returns the fully legal string key used for cache systems
+func cacheKey(k *datastore.Key) string {
```

Comment:
Why do you have this function private?
I use `MemcacheKey` to delete inconsistent negative cache (It looks happen concurrent get / put operations) with earlier versions. There’s no alternate way to fo that.

Reply:
Thanks for your feedback!

It's private because everything that is made public needs to be kept public and working forever due to pre-modules compatibility guarantees. However like I mentioned in this PR, if knowing the final cache key is something that people want (and it seems you do want it!) then that's something worth trying to do.

I have opened a new issue (#93) to track this and I hope to have some sort of solution before September.